sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/ResourceDetectorFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.fileconfig;
+
+import static java.util.stream.Collectors.joining;
+
+import io.opentelemetry.api.incubator.config.DeclarativeConfigException;
+import io.opentelemetry.sdk.autoconfigure.internal.SpiHelper;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.ExperimentalResourceDetectorModel;
+import io.opentelemetry.sdk.resources.Resource;
+import java.io.Closeable;
+import java.util.List;
+import java.util.Map;
+
+final class ResourceDetectorFactory
+    implements Factory<ExperimentalResourceDetectorModel, Resource> {
+
+  private static final ResourceDetectorFactory INSTANCE = new ResourceDetectorFactory();
+
+  private ResourceDetectorFactory() {}
+
+  static ResourceDetectorFactory getInstance() {
+    return INSTANCE;
+  }
+
+  @Override
+  public Resource create(
+      ExperimentalResourceDetectorModel model, SpiHelper spiHelper, List<Closeable> closeables) {
+    Map<String, Object> additionalProperties = model.getAdditionalProperties();
+    if (!additionalProperties.isEmpty()) {
+      if (additionalProperties.size() > 1) {
+        throw new DeclarativeConfigException(
+            "Invalid configuration - multiple resource detectors set: "
+                + additionalProperties.keySet().stream().collect(joining(",", "[", "]")));
+      }
+      Map.Entry<String, Object> detectorKeyValue =
+          additionalProperties.entrySet().stream()
+              .findFirst()
```

Comment:
Since you're already in stream/optional functional land, I think it's cleaner to return the chained result directly by leveraging `map()`:

```
return additionalProperties.entrySet().stream()
              .findFirst()
              .map(detectorKeyValue -> FileConfigUtil.loadComponent(
                  spiHelper, Resource.class, detectorKeyValue.getKey(), detectorKeyValue.getValue()))
              .orElseThrow(
                  () ->
                      new IllegalStateException("Missing detector. This is a programming error."));
```

Reply:
Surprisingly, after applying spotless, the full stream / optional approach is 11 lines, vs. 8 lines without.

I think what we need is a helper function which accepts a `Map<String, Object>`, asserts that it has exactly 1 entry, and if yes, extracts that entry returning `Map.Entry<String, Object>`. This pattern shows up in several places and would clean all of them up.